pkgs/by-name/sa/saunafs/package.nix

```diff
@@ -0,0 +1,74 @@
+{
+  lib,
+  gcc11Stdenv,
+  fetchFromGitHub,
+  cmake,
+  pkg-config,
+  acl,
+  boost175,
+  bzip2,
+  db,
+  fmt_10,
+  fuse3,
+  judy,
+  linux-pam,
+  spdlog,
+  xz,
+  yaml-cpp,
+  zlib,
+  zstd,
+}:
+
+gcc11Stdenv.mkDerivation rec {
+  pname = "saunafs";
+  version = "4.0.1";
+
+  src = fetchFromGitHub {
+    owner = "leil-io";
+    repo = "saunafs";
+    rev = "v${version}";
+    hash = "sha256-R9a0AfG/SS95TQVB34fst05cZpmqupWW/XN+1ehbTf8=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
+
+  buildInputs = [
+    (boost175.override { stdenv = gcc11Stdenv; })
+    (yaml-cpp.override { stdenv = gcc11Stdenv; })
```

Comment:
Can you provide a comment why this override is needed?

Reply:
Because of `gcc11Stdenv.mkDerivation`, without overriding stdenv, linking errors will be thrown

Reply:
There is an assumption that saunafs was written for Ubuntu 18, which uses the old gcc. Since gcc 12 has some compatibility issues with legacy code, it is necessary to use the old gcc for saunafs.

Example gcc11<->gcc12 incompatibility - some operations are no longer considered constexpr

https://github.com/Kamillaova/yanet-flake/blob/mistress/pkgs/yanet/0003-Fix-building-on-new-compilers.patch#L11

Reply:
You can see ther examples of fixing issues like this here:

https://github.com/yanet-platform/yanet/compare/main...GeorgyKirichenko:yanet:fix-compilation

Reply:
In YANET, before the fixes, gcc 11 (or older) was also needed
![image](https://github.com/NixOS/nixpkgs/assets/54859825/b060563d-b402-45c1-9c2d-ccbfe20d9b43)